hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -2014,7 +2202,7 @@ abstract class NDArrayEmitter(
    val outputElementPType: PType,
    val setup: Code[_]) {
 
-  def outputElement(idxVars: Seq[ClassFieldRef[Long]]): Code[_]
+  def outputElement(idxVars: Seq[Settable[Long]]): Code[_]
```

Comment:
I can go back to `ClassFieldRef`s. I got a comment at some point about making a lot of class fields so I was trying to avoid making tons of class fields by using locals. I don't understand tradeoffs here in general (though what you're saying about it getting split makes sense).